components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/store/UserSessionStore.java

```diff
@@ -729,4 +729,32 @@ public void removeFederatedAuthSessionInfo(String sessionContextKey) throws User
                     "the session index:" + sessionContextKey, e);
         }
     }
+
+    /**
+     * Method to check whether the user id is available in the IDN_AUTH_USER table.
+     *
+     * @param userId    Id of the user
+     * @return the boolean decision
+     * @throws UserSessionException if an error occurs when retrieving the mapping from the database
+     */
+    public boolean isExistingUser(String userId) throws UserSessionException {
+
+        Boolean isExisting = false;
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection(false)) {
+            try (PreparedStatement preparedStatement = connection
+                    .prepareStatement(SQLQueries.SQL_SELECT_INFO_OF_USER_ID)) {
+                preparedStatement.setString(1, userId);
+                try (ResultSet resultSet = preparedStatement.executeQuery()) {
+                    if (resultSet.next()) {
+                        isExisting = true;
+                    }
+                }
+            } catch (SQLException e1) {
```

Comment:
Since this catch block is the same as the outer one, let's remove it.

Reply:
Changed

Reply:
Changed